### ISSUE TITLE
Lack of C++11 support appears to not be determined when inside AmberTools

### ISSUE THREAD

**Issue by drroe** [Amber-MD/cpptraj#743]
User reported issues with Intel compilers v 16 on top of GCC 4.4.7 when trying to build inside AmberTools:
```
CXX Ewald.cpp
CXX Ewald_ParticleMesh.cpp
helpme_standalone.h(2133): error: more than one instance of overloaded function "std::to_string" matches the argument list:
            function "std::to_string(long long)"
            function "std::to_string(unsigned long long)"
            function "std::to_string(long double)"
            argument types are: (int)
              msg += std::to_string(derivativeLevel + 2);
                     ^
          detected during:
```
etc.

**Comment by drroe** [comment]
Some more C++11 related compile failures:
```
CXX Ewald_ParticleMesh.cpp
helpme_standalone.h(991): internal error: assertion failed: node_has_side_effects: bad node kind (shared/cfe/edgcpfe/il.c, line 20802)

          for (unsigned int val = 0; val < nRows_; ++val) eigenPairs.push_back({eigenValues[val][0], unsortedEigenVectors[val]});
                                                          ^

compilation aborted for Ewald_ParticleMesh.cpp (code 4)
make[1]: *** [Ewald_ParticleMesh.o] Error 4
make[1]: Leaving directory `/home/guest/amber16/AmberTools/src/cpptrajnew/src'
make: *** [install] Error 2
```

```
helpme_standalone.h(2133): error: more than one instance of overloaded function "std::to_string" matches the argument list:
            function "std::to_string(long long)"
            function "std::to_string(unsigned long long)"
            function "std::to_string(long double)"
            argument types are: (int)
              msg += std::to_string(derivativeLevel + 2);
                     ^
          detected during:
            instantiation of "void helpme::BSpline<Real>::update(short, Real, short, short) [with Real=double]" at line 2142
            instantiation of "helpme::BSpline<Real>::BSpline(short, Real, short, short) [with Real=double]" at line 2279
            instantiation of "helpme::SplineCacheEntry<Real>::SplineCacheEntry(int, int) [with Real=double]" at line 111 of "/usr/include/c++/4.4.7/ext/new_allocator.h"
            instantiation of "void __gnu_cxx::new_allocator<_Tp>::construct(__gnu_cxx::new_allocator<_Tp>::pointer, _Args &&...) [with _Tp=helpme::SplineCacheEntry<double>, _Args=<int &, int &>]" at line 96 of "/usr/include/c++/4.4.7/bits/vector.tcc"
            instantiation of "void std::vector<_Tp, _Alloc>::emplace_back(_Args &&...) [with _Tp=helpme::SplineCacheEntry<double>, _Alloc=std::allocator<helpme::SplineCacheEntry<double>>, _Args=<int &, int &>]" at line 2532
            instantiation of "void helpme::PMEInstance<Real>::filterAtomsAndBuildSplineCache(int, const helpme::PMEInstance<Real>::RealMat &) [with Real=double]" at line 4205
            instantiation of "Real helpme::PMEInstance<Real>::computeERec(int, const helpme::PMEInstance<Real>::RealMat &, const helpme::PMEInstance<Real>::RealMat &) [with Real=double]" at line 191 of "Ewald_ParticleMesh.cpp"

helpme_standalone.h(2447): error: class "std::vector<std::pair<short, short>, std::allocator<std::pair<short, short>>>" has no member "shrink_to_fit"
              splineIterator.shrink_to_fit();
                             ^
          detected during:
            instantiation of "void helpme::PMEInstance<Real>::common_init(int, Real, int, int, int, int, Real, int) [with Real=double]" at line 4429
            instantiation of "void helpme::PMEInstance<Real>::setup(int, Real, int, int, int, int, Real, int) [with Real=double]" at line 179 of "Ewald_ParticleMesh.cpp"

helpme_standalone.h(2451): error: class "std::vector<std::vector<std::pair<short, short>, std::allocator<std::pair<short, short>>>, std::allocator<std::vector<std::pair<short, short>, std::allocator<std::pair<short, short>>>>>" has no member "shrink_to_fit"
          gridIterator.shrink_to_fit();
                       ^
          detected during:
            instantiation of "void helpme::PMEInstance<Real>::common_init(int, Real, int, int, int, int, Real, int) [with Real=double]" at line 4429
            instantiation of "void helpme::PMEInstance<Real>::setup(int, Real, int, int, int, int, Real, int) [with Real=double]" at line 179 of "Ewald_ParticleMesh.cpp"
```

**Comment by drroe** [comment]
The problem in these cases is probably that the GNU backend is too old and not fully C++11 compliant. Intel compilers use a large part of the GNU infrastructure (see https://software.intel.com/en-us/cpp-compiler-developer-guide-and-reference-gcc-compatibility-and-interoperability). I think the oldest version of GNU that is fully C++11 compliant is 4.8.1 (see https://gcc.gnu.org/projects/cxx-status.html#cxx11).

**Comment by drroe** [comment]
Should be addressed by #760.
